azure-pipelines.yml

```diff
@@ -8,11 +8,6 @@ jobs:
 - job:
   strategy:
     matrix:
-      linux:
```

Comment:
Why are we removing this? Just because it's mostly redundant the ASAN and serialization jobs?

Reply:
This new job runs the native tests, so I think the existing linux job is essentially redundant. 

At some point we could probably also consolidate some of the other jobs (thanks to https://github.com/TileDB-Inc/TileDB/pull/1499).

Reply:
I guess my question is: do we not see value in native testing on both centos and ubuntu? I'm mostly concerned about differences in the filesystem since CentOS7 ships with a different FS (XFS) than Ubuntu. Although both Centos6 and Ubuntu ship with ext4 so it's not a big deal until/unless we support centos7.